# Lab Instructions

Find a dataset that interests you. I'd recommend starting on [Kaggle](https://www.kaggle.com/). Read through all of the material about the dataset and download a .CSV file.

1. Write a short summary of the data.  Where did it come from?  How was it collected?  What are the features in the data?  Why is this dataset interesting to you?  

2. Identify 5 interesting questions about your data that you can answer using Pandas methods.  

3. Answer those questions!  You may use any method you want (including LLMs) to help you write your code; however, you should use Pandas to find the answers.  LLMs will not always write code in this way without specific instruction.  

4. Write the answer to your question in a text box underneath the code you used to calculate the answer.



## 1. Dataset Summary

This dataset was downloaded from Kaggle. It contains information about the global AI and machine learning job market. The dataset includes more than 15,000 job listings from over 50 countries.

The data was created for educational and research purposes. It was generated based on industry research and real job market trends to simulate how AI jobs, salaries, and hiring patterns look around the world.

The dataset includes several features such as job title, salary in USD, experience level, employment type, company location, company size, employee residence, and remote work ratio. These features help show how different factors may affect salaries and job opportunities in the AI field.

This dataset is interesting to me because I would like to explore machine learning in the future. I am curious about how available machine learning jobs are and what factors affect salaries in this field. By analyzing this dataset, I can better understand the AI job market and the opportunities that may exist in this industry.

In [3]:
import pandas as pd

df = pd.read_csv("ai_job_dataset.csv")

df.head(10
       )

,job_id,job_title,salary_usd,salary_currency,experience_level,employment_type,company_location,company_size,employee_residence,remote_ratio,required_skills,education_required,years_experience,industry,posting_date,application_deadline,job_description_length,benefits_score,company_name
0,AI00001,AI Research Scientist,90376,USD,SE,CT,China,M,China,50,"Tableau, PyTorch, Kubernetes, Linux, NLP",Bachelor,9,Automotive,2024-10-18,2024-11-07,1076,5.9,Smart Analytics
1,AI00002,AI Software Engineer,61895,USD,EN,CT,Canada,M,Ireland,100,"Deep Learning, AWS, Mathematics, Python, Docker",Master,1,Media,2024-11-20,2025-01-11,1268,5.2,TechCorp Inc
2,AI00003,AI Specialist,152626,USD,MI,FL,Switzerland,L,South Korea,0,"Kubernetes, Deep Learning, Java, Hadoop, NLP",Associate,2,Education,2025-03-18,2025-04-07,1974,9.4,Autonomous Tech
3,AI00004,NLP Engineer,80215,USD,SE,FL,India,M,India,50,"Scala, SQL, Linux, Python",PhD,7,Consulting,2024-12-23,2025-02-24,1345,8.6,Future Systems
4,AI00005,AI Consultant,54624,EUR,EN,PT,France,S,Singapore,100,"MLOps, Java, Tableau, Python",Master,0,Media,2025-04-15,2025-06-23,1989,6.6,Advanced Robotics
5,AI00006,AI Architect,123574,EUR,SE,CT,Germany,M,Germany,50,"Data Visualization, R, SQL, Linux",Associate,7,Healthcare,2024-08-31,2024-10-04,819,5.9,Neural Networks Co
6,AI00007,Principal Data Scientist,79670,GBP,MI,FL,United Kingdom,S,United Kingdom,0,"R, Docker, MLOps",Associate,3,Gaming,2024-12-29,2025-02-28,1936,6.3,DataVision Ltd
7,AI00008,NLP Engineer,70640,EUR,EN,FL,France,L,France,0,"Python, SQL, Computer Vision, Java, Azure",Master,0,Healthcare,2024-06-07,2024-07-01,1286,7.6,Cloud AI Solutions
8,AI00009,Data Analyst,160710,USD,SE,CT,Singapore,L,Singapore,0,"Hadoop, Git, Mathematics, Python",PhD,7,Government,2024-11-04,2024-11-24,551,9.3,Quantum Computing Inc
9,AI00010,AI Software Engineer,102557,USD,SE,PT,Austria,M,Austria,0,"MLOps, GCP, Scala, Azure, Linux",Master,5,Government,2024-10-20,2024-11-06,2340,5.8,Cloud AI Solutions


## Question 1
### What is the average salary of AI jobs in this dataset?

In [4]:
average_salary = df['salary_usd'].mean()
average_salary

115348.96513333333

**Answer**

The average salary of AI jobs in this dataset is approximately **$115,349 per year**. This shows that careers in artificial intelligence and machine learning tend to offer relatively high salaries compared to many other industries.

## Question 2
### Which job titles appear most frequently in the dataset?

In [5]:
df['job_title'].value_counts().head(10)

job_title
Machine Learning Researcher    808
AI Software Engineer           784
Autonomous Systems Engineer    777
Machine Learning Engineer      772
AI Architect                   771
Head of AI                     765
NLP Engineer                   762
Robotics Engineer              759
Data Analyst                   759
AI Research Scientist          756
Name: count, dtype: int64

**Answer**

The most common job title in the dataset is **Machine Learning Researcher**, with **808 job postings**. Other common positions include **AI Software Engineer (784)**, **Autonomous Systems Engineer (777)**, **Machine Learning Engineer (772)**, and **AI Architect (771)**. This suggests that companies are actively hiring engineers and researchers to develop AI technologies.


## Question 3
### How does salary differ by experience level?

In [6]:
df.groupby('experience_level')['salary_usd'].mean()

experience_level
EN     63133.377084
EX    187723.647340
MI     87955.471833
SE    122187.657845
Name: salary_usd, dtype: float64

**Answer**

The analysis shows that salary increases as experience level increases.

- **Entry-level (EN):** about **$63,133**
- **Mid-level (MI):** about **$87,955**
- **Senior-level (SE):** about **$122,188**
- **Executive-level (EX):** about **$187,724**

This indicates that experience plays a major role in determining salary in the AI job market.


## Question 4
### Which countries have the highest number of AI job postings?

In [7]:
df['company_location'].value_counts().head(10)

company_location
Germany      814
Denmark      778
France       769
Canada       769
Austria      765
Singapore    764
China        763
India        754
Sweden       752
Israel       751
Name: count, dtype: int64

**Answer**

The country with the highest number of AI job postings is **Germany with 814 listings**. Other countries with many AI job opportunities include **Denmark (778)**, **France (769)**, **Canada (769)**, and **Austria (765)**. This suggests that AI job opportunities are distributed across several countries with strong technology sectors.


## Question 5
### Do remote jobs pay more or less than non-remote jobs?

In [8]:
df.groupby('remote_ratio')['salary_usd'].mean()

remote_ratio
0      114140.231133
50     115776.806793
100    116160.545935
Name: salary_usd, dtype: float64

**Answer**

The results show that salaries are relatively similar across different remote work levels.

- **On-site jobs (0% remote):** about **$114,140**
- **Hybrid jobs (50% remote):** about **$115,777**
- **Fully remote jobs (100% remote):** about **$116,161**

This suggests that remote work does not significantly reduce salaries in AI jobs, and fully remote jobs may even offer slightly higher salaries on average.